scipy/optimize/tests/test_optimize.py

```diff
@@ -226,6 +226,41 @@ def test_powell(self):
                          [1.72949016, -0.44156936, 0.47576729]],
                         atol=1e-14, rtol=1e-7)
 
+    def test_powell_bounded(self):
+        # Powell (direction set) optimization routine
+        # same as test_powell above, but with bounds
+        bounds = [(-np.pi, np.pi) for _ in self.startparams]
+        if self.use_wrapper:
+            opts = {'maxiter': self.maxiter, 'disp': self.disp,
+                    'return_all': False}
+            res = optimize.minimize(self.func, self.startparams, args=(),
+                                    bounds=bounds, 
+                                    method='Powell', options=opts)
+            params, fopt, direc, numiter, func_calls, warnflag = (
+                    res['x'], res['fun'], res['direc'], res['nit'],
+                    res['nfev'], res['status'])
+        else:
+            retval = optimize.fmin_powell(self.func, self.startparams,
+                                          args=(), bounds=bounds,
+                                          maxiter=self.maxiter,
+                                          full_output=True, disp=self.disp,
+                                          retall=False)
+            (params, fopt, direc, numiter, func_calls, warnflag) = retval
+
+        assert_allclose(self.func(params), self.func(self.solution),
+                        atol=1e-6)
+
+        # Ensure that function call counts are 'known good'.
+        #
+        # However, some leeway must be added: the exact evaluation
+        # count is sensitive to numerical error, and floating-point
+        # computations are not bit-for-bit reproducible across
+        # machines, and when using e.g. MKL, data alignment
+        # etc. affect the rounding error.
+        #
+        assert_(self.funccalls <= 131 + 20, self.funccalls)
```

Comment:
Which of these numbers is the leeway? Could small changes in the CI suite make this insufficient leeway in the future?

Reply:
Whenever I run it locally, I find exactly 131 funccalls. However, without the `+20` leeway, one of Travis-CI checks (TESTMODE=fast COVERAGE=NUMPYSPEC="--upgrade numpy" MB_PYTHON_VERSION=3.7), it finds 138 funccalls. All the rest of the Travis-CI checks find 131 funccalls. I am not sure why this is, but this leeway was also included in the `test_powell` function (right above where I put my `test_bounded_powell` function). Whoever included that test also included a 20 funccall leeway. I just did as they did.

Reply:
OK. It would be great if you would explain these magic numbers in the comments.

Reply:
Done.